pkg/apis/extensions/types.go

```diff
@@ -82,43 +81,45 @@ type SubresourceReference struct {
 	Subresource string `json:"subresource,omitempty"`
 }
 
-// ResourceConsumption is an object for specifying average resource consumption of a particular resource.
-type ResourceConsumption struct {
-	Resource api.ResourceName  `json:"resource,omitempty"`
-	Quantity resource.Quantity `json:"quantity,omitempty"`
+type Utilization float64
+
+type CPUTargetUtilization struct {
+	// Utilization target defines fraction of the requested CPU that should be utilized/used,
+	// e.g. 0.7 means that 70% of the requested CPU should be in use.
+	UtilizationTarget Utilization `json:"utilizationTarget"`
 }
 
 // HorizontalPodAutoscalerSpec is the specification of a horizontal pod autoscaler.
 type HorizontalPodAutoscalerSpec struct {
 	// ScaleRef is a reference to Scale subresource. HorizontalPodAutoscaler will learn the current resource consumption from its status,
 	// and will set the desired number of pods by modyfying its spec.
-	ScaleRef *SubresourceReference `json:"scaleRef"`
+	ScaleRef SubresourceReference `json:"scaleRef"`
 	// MinReplicas is the lower limit for the number of pods that can be set by the autoscaler.
-	MinReplicas int `json:"minReplicas"`
+	MinReplicas *int `json:"minReplicas"`
 	// MaxReplicas is the upper limit for the number of pods that can be set by the autoscaler. It cannot be smaller than MinReplicas.
 	MaxReplicas int `json:"maxReplicas"`
-	// Target is the target average consumption of the given resource that the autoscaler will try to maintain by adjusting the desired number of pods.
-	// Currently two types of resources are supported: "cpu" and "memory".
-	Target ResourceConsumption `json:"target"`
+	// CPU utilization is the target average CPU utilization (represented as a fraction of requested CPU) over all the pods.
+	CPUUtilization *CPUTargetUtilization `json:"cpuUtilization,omitempty"`
```

Comment:
I still prefer a map here in the long run.

As a JS user this seems really verbose:

```
spec.cpuUtilization.utilizationTarget
```

versus

```
spec.target[cpu] = 99
```

Can we stub how memory would have looked if/when we go add it at a later time?